internal/worker/providertracker/nontrackedworker.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package providertracker
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v4"
+	"gopkg.in/tomb.v2"
+
+	"github.com/juju/juju/core/logger"
+	coremodel "github.com/juju/juju/core/model"
+	"github.com/juju/juju/environs/cloudspec"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/internal/uuid"
+)
+
+// NonTrackedConfig is a struct that contains the necessary information to
+// create a non-tracked worker.
+type NonTrackedConfig struct {
+	// ModelType is the type of the model.
+	ModelType coremodel.ModelType
+
+	// ModelConfig is the model configuration for the provider.
+	ModelConfig *config.Config
```

Comment:
Do we actually need the full config? looks like `ModelType` is enough right?

Reply:
We need the full model config to create an "environ" provider.

https://github.com/juju/juju/blob/569f53ce5c183acba4dae8b0d06056f53943b2e8/environs/environ.go#L51-L72

Reply:
Oh yes of course, I missed it. Thanks